### PR TITLE
fix(PE-8765): override eval handler and add remove-controller flag on transfer

### PR DESCRIPTION
**Pull request by atticusofsparta**
## Summary by CodeRabbit

* **New Features**
  * Added Remove-Controllers tag support for transfer operations; set Remove-Controllers=false to preserve controllers during transfer.

* **Security**
  * Disabled Eval functionality to enhance process security.

* **Tests**
  * Updated test suites to validate new controller removal behavior and Eval disablement.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/common/balances.lua [hunk 1]
```diff
@@ -10,14 +10,17 @@ local balances = {}
 
 --- Transfers the ANT to a specified wallet.
 ---@param to string - The wallet address to transfer the balance to.
+---@param removeControllers boolean - Whether to remove all controllers.
 ---@param allowUnsafeAddresses AllowUnsafeAddresses
 ---@return table<string, integer>
-function balances.transfer(to, allowUnsafeAddresses)
+function balances.transfer(to, removeControllers, allowUnsafeAddresses)
```
Comment (arielmelendez) [inline src/common/balances.lua:16]: A little scary as now every call to transfer has to be reviewed for having the correct parameters passed through (or, if not present, defaults are still acceptable).

File: test/balances.test.mjs [hunk 2]
```diff
@@ -230,4 +230,314 @@ describe('aos Balances', async () => {
...
+      // Verify owner changed
+      const infoAfter = await getInfo(transferResult.Memory);
+      assert.strictEqual(
+        infoAfter.Owner,
+        STUB_NEW_OWNER,
+        'Owner should be updated to new owner',
+      );
+    });
+
+    it('should remove controllers when Remove-Controllers value is empty string (defaults to true)', async () => {
```
Comment (arielmelendez) [inline test/balances.test.mjs:365]: Nit: could add an additional test for passing any other string than true/false/empty (which should behave like true and empty).

**Review by arielmelendez — APPROVED** [review]
